web/app/widgets/chart/chart.widget.js

```diff
@@ -76,23 +76,27 @@
         };
 
         if (vm.widget.charttype == 'interactive') {
-            var startDate = new Date();
-            switch (vm.widget.period)
-            {
-                case 'h': startDate.setTime(startDate.getTime() - 60*60*1000); break;
-                case '4h': startDate.setTime(startDate.getTime() - 4*60*60*1000); break;
-                case '8h': startDate.setTime(startDate.getTime() - 8*60*60*1000); break;
-                case '12h': startDate.setTime(startDate.getTime() - 12*60*60*1000); break;
-                case 'D': startDate.setTime(startDate.getTime() - 24*60*60*1000); break;
-                case '3D': startDate.setTime(startDate.getTime() - 3*24*60*60*1000); break;
-                case 'W': startDate.setTime(startDate.getTime() - 7*24*60*60*1000); break;
-                case '2W': startDate.setTime(startDate.getTime() - 2*7*24*60*60*1000); break;
-                case 'M': startDate.setTime(startDate.getTime() - 31*24*60*60*1000); break; //Well...
-                case '2M': startDate.setTime(startDate.getTime() - 2*31*24*60*60*1000); break;
-                case '4M': startDate.setTime(startDate.getTime() - 4*31*24*60*60*1000); break;
-                case 'Y': startDate.setTime(startDate.getTime() - 12*31*24*60*60*1000); break;
-                default: startDate.setTime(startDate.getTime() - 24*60*60*1000); break;
+            function startTime() {
+            	var startDate = new Date();
+                switch (vm.widget.period)
+                {
+                    case 'h': startDate.setTime(startDate.getTime() - 60*60*1000); break;
+                    case '4h': startDate.setTime(startDate.getTime() - 4*60*60*1000); break;
+                    case '8h': startDate.setTime(startDate.getTime() - 8*60*60*1000); break;
+                    case '12h': startDate.setTime(startDate.getTime() - 12*60*60*1000); break;
+                    case 'D': startDate.setTime(startDate.getTime() - 24*60*60*1000); break;
+                    case '3D': startDate.setTime(startDate.getTime() - 3*24*60*60*1000); break;
+                    case 'W': startDate.setTime(startDate.getTime() - 7*24*60*60*1000); break;
+                    case '2W': startDate.setTime(startDate.getTime() - 2*7*24*60*60*1000); break;
+                    case 'M': startDate.setTime(startDate.getTime() - 31*24*60*60*1000); break; //Well...
+                    case '2M': startDate.setTime(startDate.getTime() - 2*31*24*60*60*1000); break;
+                    case '4M': startDate.setTime(startDate.getTime() - 4*31*24*60*60*1000); break;
+                    case 'Y': startDate.setTime(startDate.getTime() - 12*31*24*60*60*1000); break;
+                    default: startDate.setTime(startDate.getTime() - 24*60*60*1000); break;
+                }
+                return startDate;
             }
+            var startDate = startTime();
```

Comment:
Yep, the start date is used initially when chart is initialized. When an update is received the timespan must be recalculated based on the current time.